pageserver/src/virtual_file.rs

```diff
@@ -173,55 +173,162 @@ impl OpenFiles {
     }
 }
 
-#[derive(Debug, thiserror::Error)]
-pub enum CrashsafeOverwriteError {
-    #[error("final path has no parent dir")]
-    FinalPathHasNoParentDir,
-    #[error("remove tempfile: {0}")]
-    RemovePreviousTempfile(#[source] std::io::Error),
-    #[error("create tempfile: {0}")]
-    CreateTempfile(#[source] std::io::Error),
-    #[error("write tempfile: {0}")]
-    WriteContents(#[source] std::io::Error),
-    #[error("sync tempfile: {0}")]
-    SyncTempfile(#[source] std::io::Error),
-    #[error("rename tempfile to final path: {0}")]
-    RenameTempfileToFinalPath(#[source] std::io::Error),
-    #[error("open final path parent dir: {0}")]
-    OpenFinalPathParentDir(#[source] std::io::Error),
-    #[error("sync final path parent dir: {0}")]
-    SyncFinalPathParentDir(#[source] std::io::Error),
+const ENOSPC: i32 = 28;
```

Comment:
```suggestion
const ENOSPC: i32 = nix::errno:Errno::ENOSPC as i32;
```

Reply:
(you can also convert it to `nix::errno::Errno` in the match and avoid having the constants altogether)

Reply:
Thanks, when I was preparing this draft I couldn't remember which crate it was that provided errno constants.  Updated this to use nix types in 0a502f411756491775499422d2c60dee75208273